### ISSUE TITLE
Able to create a password without a numer or an upper case character

### ISSUE THREAD

**Issue by roman87** [TeamMentor/TM_4_0_Design#259]
We need to fix both the Jade frontend and the 3.5 backend to prevent users from creating passwords which do not match our policy. In this case we should prevent creation and give a proper error for passwords which do not contain an upper case letter, a number and a special character and are less then 8 characters long. 
Related to https://github.com/TeamMentor/Master/issues/932

**Comment by michaelhidalgo** [comment]
I'm following the recommendation from TEAM Mentor article https://teammentor.net/article/60802624-5588-4c8f-aac3-baa27cd28187

In this article,  the SqlMembershipProvider is explained.

**Comment by roman87** [comment]
The recommendation looks good, but lets write a test to validate it :)

**Comment by michaelhidalgo** [comment]
So I came up with this RegEx which seems to work fine (((?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*\W).{8,256}))

Here are some unit test : 


![screen shot 2015-01-19 at 3 40 16 pm](https://cloud.githubusercontent.com/assets/1732035/5808573/90d9d086-9ff4-11e4-808f-93fa1570ae2d.png)

**Comment by DinisCruz** [comment]
Do we have tests that stress test this regex?

We don't want another instance of http://blog.diniscruz.com/2013/03/the-email-regex-that-could-had-dosed.html

**Comment by michaelhidalgo** [comment]
Good point. Well there is a length check in the first place ( between 8 and 256 characters). However, what fuzzing technique would you guys recommend :)?

**Comment by DinisCruz** [comment]
I think the best will be to write a unit test like the one you can see on that blog post (which is now part of TM 3.5 codebase :) )

**Comment by michaelhidalgo** [comment]
Yeah I agree. that test scenario will be covered when changing the validation in the back end. In the mean time, I wrote up a quick test to fuzz it : 
A few notes : 

1. It was tested 10,000 times ( in a loop)
2. The size of the payload was 1012 characters randomly generated (Thanks FluentSharp)

![screen shot 2015-01-20 at 8 27 07 am](https://cloud.githubusercontent.com/assets/1732035/5818888/daa11924-a07e-11e4-915c-4273d7dc84ca.png)

This is the results : 

![screen shot 2015-01-20 at 8 34 18 am](https://cloud.githubusercontent.com/assets/1732035/5818934/27e3abca-a07f-11e4-9d86-9face33ca0ba.png)

I did see a few CPU spikes but they are acceptable considering the number of iterations.
